### PR TITLE
Add release notes for  1.3.1

### PR DESCRIPTION
**Pull request by gaiksaya**
Signed-off-by: Sayali Gaikawad <gaiksaya@amazon.com>

### Description
Adds release notes for 1.3.1
 
### Issues Resolved
part of https://github.com/opensearch-project/opensearch-build/issues/1805
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-release-notes-1.3.1.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+## OpenSearch Security
+### Bug fixes
+
+* Fix for "java.lang.IllegalArgumentException: The array of keys must not be null" for "_cat/health" requests ([#1048](https://github.com/opendistro-for-elasticsearch/security/pull/1048))
+* Revert "Fix AuthCredentials equality (#876)" to improve performance ([#1061](https://github.com/opendistro-for-elasticsearch/security/pull/1061))
+
+### Maintenance
+
+* Bump version to 1.13.1.0 ([#1054](https://github.com/opendistro-for-elasticsearch/security/pull/1054))
```
Comment (stockholmux) [inline release-notes/opensearch-release-notes-1.3.1.md:11]: Why is there a couple of ODFE PRs in this?

**Review by stockholmux — COMMENTED** [review]
Confusing linking to ODFE

File: release-notes/opensearch-release-notes-1.3.1.md (hunk 1 above)
Comment (gaiksaya) [reply release-notes/opensearch-release-notes-1.3.1.md:11]: @cliu123 might be able to answer that

Comment (cliu123) [reply release-notes/opensearch-release-notes-1.3.1.md:11]: That is security plugin ODFE 1.13 release note. The OpenSearch 1.3.1 release note is here: https://github.com/opensearch-project/security/blob/1.3/release-notes/opensearch-security.release-notes-1.3.1.0.md

Comment (gaiksaya) [reply release-notes/opensearch-release-notes-1.3.1.md]: Thanks @cliu123. Fixed the release notes and format too.
cc: @peterzhuamazon

**Review by bbarani — APPROVED** [review]

**Review by peterzhuamazon — APPROVED** [review]
Thanks.

**Pull request merged**